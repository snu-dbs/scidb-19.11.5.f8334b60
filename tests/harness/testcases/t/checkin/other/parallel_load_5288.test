# SDB-5288
# parallel binary save/input.

--setup
--start-query-logging
--test

# nullable without null values
save(build(<v:double>[i=0:3,2,0, j=0:3,2,0], i*2+j), '5288_tmp_data_nullable_without_null.bin', -1, '(double null)')
sort(input(<v:double>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_nullable_without_null.bin', -1, '(double null)'))

# nullable with null values
save(build(<v:double>[i=0:3,2,0, j=0:3,2,0], iif(i=1, null, i*2+j)), '5288_tmp_data_nullable_with_null.bin', -1, '(double null)')
sort(input(<v:double>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_nullable_with_null.bin', -1, '(double null)'))

# not nullable
save(build(<v:double not null>[i=0:3,2,0, j=0:3,2,0], i*2+j), '5288_tmp_data_not_nullable.bin', -1, '(double)')
sort(input(<v:double not null>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_not_nullable.bin', -1, '(double)'))

# All the above are for binary format.
# Try opaque, with nullable with null values
save(build(<v:double>[i=0:3,2,0, j=0:3,2,0], iif(i=1, null, i*2+j)), '5288_tmp_data_nullable_with_null.opa', -1, 'opaque')
sort(input(<v:double>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_nullable_with_null.opa', -1, 'opaque'))

# Opaque works with sparse array, too. (Note that binary does not!)
save(filter(build(<v:double>[i=0:3,2,0, j=0:3,2,0], i*2+j), i=j), '5288_tmp_data_sparse.opa', -1, 'opaque')
sort(input(<v:double>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_sparse.opa', -1, 'opaque'))

--cleanup
--shell --command "find ${SCIDB_DATA_PATH} -name '5288_tmp_data*' | xargs rm -f"
