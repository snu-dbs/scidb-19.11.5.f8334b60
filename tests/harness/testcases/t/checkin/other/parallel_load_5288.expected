SCIDB QUERY : <save(build(<v:double>[i=0:3,2,0, j=0:3,2,0], i*2+j), '5288_tmp_data_nullable_without_null.bin', -1, '(double null)')>
{i,j} v

SCIDB QUERY : <sort(input(<v:double>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_nullable_without_null.bin', -1, '(double null)'))>
{$n} v
{0} 0
{1} 1
{2} 2
{3} 2
{4} 3
{5} 3
{6} 4
{7} 4
{8} 5
{9} 5
{10} 6
{11} 6
{12} 7
{13} 7
{14} 8
{15} 9

SCIDB QUERY : <save(build(<v:double>[i=0:3,2,0, j=0:3,2,0], iif(i=1, null, i*2+j)), '5288_tmp_data_nullable_with_null.bin', -1, '(double null)')>
{i,j} v

SCIDB QUERY : <sort(input(<v:double>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_nullable_with_null.bin', -1, '(double null)'))>
{$n} v
{0} null
{1} null
{2} null
{3} null
{4} 0
{5} 1
{6} 2
{7} 3
{8} 4
{9} 5
{10} 6
{11} 6
{12} 7
{13} 7
{14} 8
{15} 9

SCIDB QUERY : <save(build(<v:double not null>[i=0:3,2,0, j=0:3,2,0], i*2+j), '5288_tmp_data_not_nullable.bin', -1, '(double)')>
{i,j} v

SCIDB QUERY : <sort(input(<v:double not null>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_not_nullable.bin', -1, '(double)'))>
{$n} v
{0} 0
{1} 1
{2} 2
{3} 2
{4} 3
{5} 3
{6} 4
{7} 4
{8} 5
{9} 5
{10} 6
{11} 6
{12} 7
{13} 7
{14} 8
{15} 9

SCIDB QUERY : <save(build(<v:double>[i=0:3,2,0, j=0:3,2,0], iif(i=1, null, i*2+j)), '5288_tmp_data_nullable_with_null.opa', -1, 'opaque')>
{i,j} v

SCIDB QUERY : <sort(input(<v:double>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_nullable_with_null.opa', -1, 'opaque'))>
{$n} v
{0} null
{1} null
{2} null
{3} null
{4} 0
{5} 1
{6} 2
{7} 3
{8} 4
{9} 5
{10} 6
{11} 6
{12} 7
{13} 7
{14} 8
{15} 9

SCIDB QUERY : <save(filter(build(<v:double>[i=0:3,2,0, j=0:3,2,0], i*2+j), i=j), '5288_tmp_data_sparse.opa', -1, 'opaque')>
{i,j} v

SCIDB QUERY : <sort(input(<v:double>[i=0:*,2,0, j=0:3,2,0], '5288_tmp_data_sparse.opa', -1, 'opaque'))>
{$n} v
{0} 0
{1} 3
{2} 6
{3} 9

